app/backend/src/couchers/db.py

```diff
@@ -29,10 +33,30 @@ def apply_migrations():
         os.chdir(cwd)
 
 
+__engine = None
+__Session = None
+
+
+def get_engine():
+    global __engine
+
+    if not __engine:
+        if config.config["IN_TEST"]:
+            __engine = create_engine(config.config["DATABASE_CONNECTION_STRING"], poolclass=NullPool)
+        else:
+            __engine = create_engine(config.config["DATABASE_CONNECTION_STRING"])
+
+    return __engine
```

Comment:
This can be written much shorter and concise with
```suggestion
@functools.cache
def get_engine():
    if config.config["IN_TEST"]:
        return create_engine(config.config["DATABASE_CONNECTION_STRING"], poolclass=NullPool)
    else:
        return create_engine(config.config["DATABASE_CONNECTION_STRING"])
```
`functools.lru_cache(-1)`works with older Pythons

Reply:
Yeah, that's really nice, didn't know that existed!